server/catgenome/src/main/java/com/epam/catgenome/entity/gene/GeneFilterForm.java

```diff
@@ -0,0 +1,181 @@
+/*
+ * MIT License
+ *
+ * Copyright (c) 2016-2021 EPAM Systems
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+package com.epam.catgenome.entity.gene;
+
+import com.epam.catgenome.dao.index.FeatureIndexDao;
+import com.epam.catgenome.dao.index.field.IndexSortField;
+import com.epam.catgenome.entity.AbstractFilterForm;
+import com.epam.catgenome.entity.index.FeatureType;
+import lombok.Getter;
+import lombok.Setter;
+import org.apache.lucene.document.IntPoint;
+import org.apache.lucene.index.Term;
+import org.apache.lucene.search.BooleanClause;
+import org.apache.lucene.search.BooleanQuery;
+import org.apache.lucene.search.PrefixQuery;
+import org.apache.lucene.search.Sort;
+import org.apache.lucene.search.SortField;
+import org.apache.lucene.search.TermQuery;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import static java.util.Collections.emptyList;
+import static org.apache.commons.collections.CollectionUtils.isEmpty;
+import static org.apache.commons.collections.CollectionUtils.isNotEmpty;
+
+@Getter
+@Setter
+public class GeneFilterForm extends AbstractFilterForm {
+    private Integer startIndex;
```

Comment:
do we need use here page? as we use it for VCF form

Reply:
@SilinPavel, I asked Maria, she said that this feature  (page) is deprecated and we should use only pageSize and Pointer, as it always will be retrieved from UI for genes.

Reply:
Okey, but in this case I see, that we lost pointer from a response, when you merge your response from lucene with bookmarks in `searchFeaturesByReference` you do not include a pointer to the new "merged" object. However `bookmarks` is deprecated, as I can remember, and I believe we can get rid of merging here at all. @mzueva what do you think?

Reply:
Done!